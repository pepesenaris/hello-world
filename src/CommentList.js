import React from 'react';

const Comment = ({ text }) => {
  return (
    <article>
      {text}
    </article>
  )
}

/*
 PropTypes are used to statically check props passed to a component
 so we can detect missing or wrongly used properties.
 They throw a warning in development enviroment to notify errors
*/

Comment.propTypes = {
  text: React.PropTypes.string.isRequired,
}

const CommentList = (props) => {
  const commentList = props.comments.map((comment) => (
    <Comment key={`comment-${comment.id}`} {...comment}/>
  ));

  /*
    Passing down a spread object to a component is a
    common pattern that saves a bit of typing and ease
    adding new props to a component in the future
  */

  return (
    <div>
      { commentList.length > 0 ? commentList : 'No comments for today' }
    </div>
  )
}

CommentList.propTypes = {
  comments: React.PropTypes.array.isRequired,
};

export default CommentList;
