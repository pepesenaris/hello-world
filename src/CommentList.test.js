import React from 'react';
import CommentList from './CommentList';
import renderer from 'react-test-renderer';

const comments = [{ text: 'Comment 1', id: 1}, { text: 'Comment 2', id: 2}];

test('it renders a list of comments', () => {
  const component = renderer.create(
    <CommentList comments={comments}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders message when no comments', () => {
  const component = renderer.create(
    <CommentList comments={[]}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
