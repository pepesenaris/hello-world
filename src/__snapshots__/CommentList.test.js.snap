exports[`test it renders a list of comments 1`] = `
<div>
  <article>
    Comment 1
  </article>
  <article>
    Comment 2
  </article>
</div>
`;

exports[`test it renders message when no comments 1`] = `
<div>
  No comments for today
</div>
`;
